### PR TITLE
Add type tests

### PR DESCRIPTION
**Pull request by aminya**
Here I added an example test. But it is more about testing the usage rather than the interface itself.

~~Also, made `timeReporter` optional~~

### CODE REVIEW

**Comment by UziTech** [comment]
I made `timeReporter` optional in https://github.com/UziTech/atom-jasmine3-test-runner/commit/9bbc77545dcb704326c78a160751a5111f80eb2f

**Comment by UziTech** [comment]
I would be ok with changing the codebase to typescript. Or maybe we could use https://github.com/ai/check-dts for tests.

**Comment by aminya** [comment]
Rebased on master.

**Comment by aminya** [comment]
> I would be ok with changing the codebase to typescript. Or maybe we could use [ai/check-dts](https://github.com/ai/check-dts) for tests.

I don't think that is necessary. `@types/` packages basically use `tsc --noEmit` to somehow verify the types against the examples of that package.

**Comment by aminya** [comment]
The CI fails to find the tsconfig. Maybe use `tsc -p ./tsconfig.json --noEmit`

File: types/index-spec.ts [hunk 1]
```diff
@@ -30,17 +28,19 @@ const optionalConfigurationFunction = function() {
 		jasmine.addMatchers({
 			toBeTheAnswerToTheUltimateQuestionOfLifeTheUniverseAndEverything: function (util, customEqualityTesters) {
 				return {
-					compare: function (actual) {
-						let result = {};
+					compare: function (actual: unknown) {
+						let result: {pass: boolean, message?: string} = {pass: true};
```
Comment (aminya) [inline types/index-spec.ts:32]: I took this from the readme, so you likely need to update the readme too.

Comment (UziTech) [reply types/index-spec.ts:32]: It does actually work without this update typescript just isn't smart enough to know that `pass` is set below this line. This change is only needed to please the typescript compiler.

**Pull request merged**